### PR TITLE
[ROCKETMQ-249] Do not attempt to clear disk even if disk store path i…

### PR DESCRIPTION
**Pull request by shroman**
…s found

https://issues.apache.org/jira/browse/ROCKETMQ-249

### CODE REVIEW

**Comment by shroman** [comment]
@vongosling @zhouxinyu @lizhanhui @vsair Anyone's willing to review the PR?

**Comment by zhouxinyu** [comment]
It seems that broker can't work properly if disk store path is not found?

**Comment by shroman** [comment]
@zhouxinyu The problem is described in the JIRA issue -- "If disk usage estimates fail because the disk store path is not found (doesn't exist), message store falsely decides the disk is full and tries to delete."
No failure, just a wasteful operation. I corrected the execution path, plus wrote the unit tests for it.

File: common/src/main/java/org/apache/rocketmq/common/UtilAll.java [hunk 1]
```diff
@@ -184,15 +184,27 @@ public static String timeMillisToHumanString3(final long t) {
...
 
+    /**
+     * Estimates the disk usage percentage by the specified path.
+     *
+     * @param path Path.
+     * @return Disk usage percentage by path.
+     */
     public static double getDiskPartitionSpaceUsedPercent(final String path) {
-        if (null == path || path.isEmpty())
+        if (null == path || path.isEmpty()) {
```
Comment (solosky) [inline common/src/main/java/org/apache/rocketmq/common/UtilAll.java]: we should use StringUtils.isBlank(..) instead as a basic practice.

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java [hunk 2]
```diff
@@ -1509,65 +1514,62 @@ private boolean isTimeToDelete() {
...
-                    }
+        /**
+         * Checks if cleaning on the disk is needed.
+         *
+         * @param usageRatio Usage ratio.
+         * @param allowedRatio Allowed ratio.
+         * @return <code>True</code> if cleaning is needed, otherwise <code>false</code>.
+         */
+        private boolean needCleaning(double usageRatio, double allowedRatio) {
+            if (usageRatio == -1)
```
Comment (dongeforever) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: It is not safe to use == for float value. In this case, usageRatio < 0 is OK

File: store/src/test/java/org/apache/rocketmq/store/CleanCommitLogServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        try {
+            messageStore.start();
+            messageStore.load();
+
+            Method m = clazz.getDeclaredMethod("isSpaceToDelete");
+            m.setAccessible(true);
+
+            Assert.assertFalse((boolean) m.invoke(cleanCommitLogService));
+        } finally {
+            messageStore.shutdown();
```
Comment (dongeforever) [inline store/src/test/java/org/apache/rocketmq/store/CleanCommitLogServiceTest.java]: How about putting the finally block into teardown() ?
The test framework will handle it.

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java (hunk 2 above)
Comment (shroman) [reply store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: Thanks, fixed.

File: store/src/test/java/org/apache/rocketmq/store/CleanCommitLogServiceTest.java (hunk 3 above)
Comment (shroman) [reply store/src/test/java/org/apache/rocketmq/store/CleanCommitLogServiceTest.java]: Good point. thanks.

**Comment by shroman** [comment]
@dongeforever Modified as you suggested, thanks!

**Comment by vongosling** [comment]
@shroman I will close the pr, if you happened to the same question, please let me know.

**Pull request closed without merging**